app/views/events/_teaser_list.html.haml

```diff
@@ -13,22 +13,23 @@
           #{t('events.show.start_date')}
       %tr
     %tbody
-      - if events.any?
-        - events.each do |event|
-          %tr.event
-            %td.event-name
-              = link_to event.name, event.sign_up_url, target: '_blank'
-            %td.location
-              = event.location
-            %td.fee.text-align-right
-              = number_to_currency(event.fee, locale: 'sv', format: '%n',
-                                   delimiter: ' ', precision: 0)
-            %td.event-start-date.text-align-center
-              = event.start_date
+      - if company.show_dinkurs_events
+        - if events.any?
+          - events.each do |event|
+            %tr.event
+              %td.event-name
+                = link_to event.name, event.sign_up_url, target: '_blank'
+              %td.location
+                = event.location
+              %td.fee.text-align-right
+                = number_to_currency(event.fee, locale: 'sv', format: '%n',
+                                     delimiter: ' ', precision: 0)
+              %td.event-start-date.text-align-center
+                = event.start_date
+        - else
+          = render 'events/empty_with_message', message: t('events.show.no_events')
```

Comment:
With the changes recommended for company `show` view, we can remove this logic (that is, we don't need or want to check `show_dinkurs_events` in this file).